Unable to download from auth-protected URLs

Encountering this issue while attempting to download a lecture. These lectures are behind a custom zoom setup for my school (could be part of the issue) where the URL begins with https://*schoolname*.zoom.etc


The following error occurs while attempting a download with a password
```
C:\Users\Gh0st\Downloads\Portable>zoomdl.exe -u "https://*redacted*" -p "gH92b$ZG"
Traceback (most recent call last):
  File "zoom_dl\__main__.py", line 20, in <module>
  File "zoom_dl\__init__.py", line 18, in main
  File "zoom_dl\zoomdl.py", line 135, in download
UnboundLocalError: local variable 'meet_id' referenced before assignment
[6720] Failed to execute script __main__
```

Current Environment is
>Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)] on win32


Also tried on WSL with no success
> Linux Gh0st-PC 4.19.84-microsoft-standard #1 SMP Wed Nov 13 11:44:37 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux


```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "./zoomdl/__main__.py", line 20, in <module>
  File "./zoomdl/zoom_dl/__init__.py", line 18, in main
  File "./zoomdl/zoom_dl/zoomdl.py", line 135, in download
UnboundLocalError: local variable 'meet_id' referenced before assignment
```

This error occurs when attempting a download which does not require a password but does require you to be logged in (in both environments.)

```No video URL in meta, going bruteforce
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "./zoomdl/__main__.py", line 20, in <module>
  File "./zoomdl/zoom_dl/__init__.py", line 18, in main
  File "./zoomdl/zoom_dl/zoomdl.py", line 142, in download
  File "./zoomdl/zoom_dl/zoomdl.py", line 80, in get_page_meta
AttributeError: 'NoneType' object has no attribute 'group'
```

Perhaps incorporate an argument to supply a cookie/auth string to enable downloads

Not sure if this should be a separate issue (seems related), but I've got a Zoom video behind a "registration" page (I enter my email, name, & captcha to view it), which I cannot download.  The script reports "[ERROR] Video not found in page. Is it login-protected?".

Are such videos not downloadable?

--- a/zoom_dl/utils.py
+++ b/zoom_dl/utils.py
@@ -70,7 +70,7 @@ def parseOpts():
                         default=1)
     PARSER.add_argument("-a", "--auth",
                         help=("Set SSO Authentication cookies _zm_kms and"
-                              "_km_page_auth cookies cookies"
+                              "_zm_ssid cookies cookies"
                               "Looks like: 'aw1_*******'"),
                         type=str,
                         default="",

--- a/zoom_dl/utils.py
+++ b/zoom_dl/utils.py
@@ -68,6 +68,13 @@ def parseOpts():
                         metavar="level",
                         type=int,
                         default=1)
+    PARSER.add_argument("-a", "--auth",
+                        help=("Authentication key _zm_kms from cookies "
+                              "cookies, for SSO protected urls."
+                              "Looks like 'aw1_*******'"),
+                        type=str,
+                        default="",
+                        metavar="auth")
     # PARSER.add_argument("-b", "--browser",
     #                     help=("Indicate the browser from which retrieve the "
     #                           "cookies, to bypass passwords and reCAPTCHA. "

--- a/zoom_dl/utils.py
+++ b/zoom_dl/utils.py
@@ -69,9 +69,9 @@ def parseOpts():
                         type=int,
                         default=1)
     PARSER.add_argument("-a", "--auth",
-                        help=("Authentication key _zm_kms from cookies "
-                              "cookies, for SSO protected urls."
-                              "Looks like 'aw1_*******'"),
+                        help=("Set SSO Authentication cookies _zm_kms and"
+                              "_km_page_auth cookies cookies"
+                              "Looks like: 'aw1_*******'"),
                         type=str,
                         default="",
                         metavar="auth")

--- a/zoom_dl/zoomdl.py
+++ b/zoom_dl/zoomdl.py
@@ -19,7 +19,7 @@ class ZoomDL():
         self.session = requests.session()
         self.loglevel = self.args.log_level
         if self.args.auth:
-            self.session.cookies.set("_zm_page_auth", self.args.auth)
+            self.session.cookies.set("_zm_ssid", self.args.auth)
             self.session.cookies.set("_zm_kms", self.args.auth)
 
     def _print(self, message, level=0):

--- a/zoom_dl/zoomdl.py
+++ b/zoom_dl/zoomdl.py
@@ -18,7 +18,9 @@ class ZoomDL():
         self.args = args
         self.session = requests.session()
         self.loglevel = self.args.log_level
-        # self._set_cookies(self.args.browser)
+        if self.args.auth:
+            self.session.cookies.set("_zm_page_auth", self.args.auth)
+            self.session.cookies.set("_zm_kms", self.args.auth)
 
     def _print(self, message, level=0):
         """Print to console, if level is sufficient.

